apps/api/src/modules/auth/auth.service.ts

```diff
@@ -1,17 +1,27 @@
 import { Injectable } from '@nestjs/common';
 import { ConfigService } from '@nestjs/config';
 import { LoginResponse } from './dto/login-response';
+import { UsersService } from '../users/users.service';
+import { comparePasswords } from 'src/common/utils/bcrypt';
 
 @Injectable()
 export class AuthService {
-  constructor(private configService: ConfigService) {}
+  constructor(
+    private configService: ConfigService,
+    private readonly usersService: UsersService,
+  ) {}
 
   async validateUser(username: string, password: string): Promise<string> {
-    const adminUsername = this.configService.getOrThrow<string>('ADMIN_USERNAME');
-    const adminPassword = this.configService.getOrThrow<string>('ADMIN_PASSWORD');
+    const entry = await this.usersService.findByUsername(username);
```

Comment:
We should handle the case where entry will be null if the user does not exist in database. Else we will get error when trying to access the username and password entries in the next two lines.

Reply:
Added check as suggested